### PR TITLE
Support python 3.11

### PR DESCRIPTION
**Pull request by maxnoe**
Python 3.11 changed the behavior of the format string for a `class Foo(str, Enum)` but also introduced a new class `StrEnum` that keeps the old behavior.

I added a `ctapipe.compat` module that encapsulates the version dependent behavior needed for ctapipe and added the previous implementation for `StrEnum` there.

Numba also needs to be at least 0.57 for python 3.11.

### CODE REVIEW

**Comment by maxnoe** [comment]
Seems we have to wait for a numba release adding python 3.11 support

**Comment by kosack** [comment]
just updated this to master - let's see if there are still issues

**Comment by maxnoe** [comment]
Yes, still no numba for 3.11

**Comment by maxnoe** [comment]
Tests are now passing using the 0.57rc1 nunba pre release

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -103,6 +107,11 @@ jobs:
         run: |
           tee -a ~/.bash_profile <<<'export PATH="$pythonLocation/bin:$PATH"'
 
+      - if: ${{ matrix.python-version == '3.11' }}
+        # install numba pre-release for python 3.11
+        run: |
+          pip install numba==0.57.0rc1
```
Comment (Tobychev) [inline .github/workflows/ci.yml:113]: Is there a way to make this "0.57rc1 or later"?

File: ctapipe/tools/apply_models.py [hunk 2]
```diff
@@ -201,7 +201,7 @@ def _apply(self, reconstructor, tel_tables, start, stop):
                 write_table(
                     table[output_columns],
                     self.output_path,
-                    f"/dl2/event/telescope/{prop}/{prefix}/tel_{tel_id:03d}",
+                    f"/dl2/event/telescope/{prop.value}/{prefix}/tel_{tel_id:03d}",
```
Comment (Tobychev) [inline ctapipe/tools/apply_models.py:204]: How is this related to the python 3.11 support?

Comment (maxnoe) [reply ctapipe/tools/apply_models.py:204]: Checking the chabgelog, there is actually  now
https://docs.python.org/3/library/enum.html#enum.StrEnum

Which might be a better solution, will change

File: .github/workflows/ci.yml (hunk 1 above)
Comment (maxnoe) [reply .github/workflows/ci.yml:113]: Yes, but I think I'd rather wait until the release is out and just remove this workaround then

**Review by Tobychev — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (maxnoe) [reply .github/workflows/ci.yml]: Numba 0.57 has been released (at least on PyPI, the conda-forge release is underway), so this was removed

**Comment by maxnoe** [comment]
I am not sure if the CI system allows for such conclusions, but: the python 3.11 test suite ran one minute faster than the python 3.10 tests (11 min vs. 10 min). Which would be a really nice improvement for just updating to a new python minor version release!

**Review by Tobychev — APPROVED** [review]

File: setup.cfg [hunk 3]
```diff
@@ -33,7 +33,7 @@ install_requires=
     importlib_metadata  ; python_version < "3.10"
     joblib
     matplotlib ~=3.0
-    numba ~=0.56.0
+    numba >=0.56
```
Comment (LukasNickel) [inline setup.cfg:36]: You wrote that we need at least numba 0.57, is that not correct anymore?

**Review by kosack — APPROVED** [review]

File: setup.cfg (hunk 3 above)
Comment (maxnoe) [reply setup.cfg:36]: For 3.11, 3.10 and below is fine with 0.56

Comment (maxnoe) [reply setup.cfg:36]: And since versions 0.56 correctly specifies that it is not compatible with 3.11, there is no issue.

The conda-forge package for 0.57 is not yet out, so prescribing 0.57 is not a good idea.
